src/robot/variables/assigner.py

```diff
@@ -205,45 +205,65 @@ def _normal_assign(self, name, value, variables):
         return value if name[0] == '$' else variables[name]
 
 
-def ReturnValueResolver(assignment):
-    if not assignment:
-        return NoReturnValueResolver()
-    if len(assignment) == 1:
-        return OneReturnValueResolver(assignment[0])
-    if any(a[0] == '@' for a in assignment):
-        return ScalarsAndListReturnValueResolver(assignment)
-    return ScalarsOnlyReturnValueResolver(assignment)
+class ReturnValueResolver:
 
+    @classmethod
+    def from_assignment(cls, assignment):
+        if not assignment:
+            return NoReturnValueResolver()
+        if len(assignment) == 1:
+            return OneReturnValueResolver(assignment[0])
+        if any(a[0] == '@' for a in assignment):
+            return ScalarsAndListReturnValueResolver(assignment)
+        return ScalarsOnlyReturnValueResolver(assignment)
 
-class NoReturnValueResolver:
+    def resolve(self, return_value):
+        raise NotImplementedError
+
+    def _split_assignment(self, assignment, handle_list_and_dict=True):
+        match: VariableMatch = search_variable(assignment, parse_type=True)
+        from robot.running import TypeInfo
```

Comment:
- Import within a method should be on the first line.
- `match` shouldn't need a type hint because `search_variable` has a return type.